client/src/AddImageLayer.js

```diff
@@ -1,7 +1,7 @@
 import React, { Component } from 'react';
 import { bindActionCreators } from 'redux';
 import { connect } from 'react-redux';
-import ActiveStorageProvider from 'react-activestorage-provider';
+import ActiveStorageProvider, { DirectUploadProvider } from 'react-activestorage-provider';
```

Comment:
Am I correct that this puts images on S3, bypassing heroku/rails altogether?

Reply:
Nope, it is identical to the ActiveStorageProvider, but it allows us to use the generated blobs directly in JS instead of automatically `PUT` requesting them to a Rails endpoint.

https://github.com/cbothner/react-activestorage-provider#directuploadprovider

I used this in order to `PUT` to multiple endpoints—a different one per image.